Fix inconsistent function signature in CDC.cpp

echtzeit-dev opened the pull request:
USBAPI.h declares the return type of Serial_::baud() as `uint32_t`

This causes an error when building the framework with Clang, because it typedefs arm-none-eabi `uint32_t` to `unsigned int` and not to `unsigned long` like gcc-arm-none-eabi.

weliveindetail committed:
Fix inconsistent function signature in CDC.cpp

USBAPI.h declares the return type of Serial_::baud() as `uint32_t`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino/ArduinoCore-sam?pullRequest=136) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino/ArduinoCore-sam?pullRequest=136) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino/ArduinoCore-sam?pullRequest=136) it.</sub>

facchinm commented:
LGTM, thanks! If you could sign the CLA I'll be glad to merge it immediately :slightly_smiling_face:

weliveindetail committed:
Fix inconsistent function signature in HardwareSerial.h

This causes an override mismatch for UARTClass::begin(const uint32_t) in UARTClass.h when compiling with Clang

weliveindetail committed:
Clang defines the long_call attribute only on MIPS

weliveindetail committed:
Honor deprecation of register keyword in C++17 when building with GCC/Clang

echtzeit-dev commented:
I have 3 more fixes. They are rather small as well. Should I make separate PRs or would you prefer a single one with 3 commits?

echtzeit-dev commented:
FYI: There appears to be lag in the CLA check. I signed it and the page behind link confirms that, but the check here is still pending (even after clicking the recheck link).

per1234 commented:
Hi @echtzeit-dev. I don't think it is a lag. The check is done against the user who created the commit. That is @weliveindetail, so it is @weliveindetail who must sign the CLA in order for the check to pass.

Hopefully the check will automatically pass once that has been done. If not, please comment here and we will investigate.

weliveindetail commented:
Thanks for you note @per1234! It worked.

echtzeit-dev commented:
> FYI: There appears to be lag in the CLA check. I signed it and the page behind link confirms that, but the check here is still pending (even after clicking the recheck link).

I added them here directly: https://github.com/arduino/ArduinoCore-sam/pull/136#commits-pushed-43df67c

echtzeit-dev commented:
@facchinm Are you still considering to merge this? Otherwise, do you have any questions or change requests? Thx
